transitclock/src/main/java/org/transitclock/utils/threading/NamedThread.java

```diff
@@ -88,16 +90,19 @@ private static String threadNameWithCounter(String name) {
 				count++;			
 			}		
 			threadNameCountMap.put(name, count);
+			l.unlock();
```

Comment:
If the threadNameCountMap.put is causing the issue, consider moving it inside the synchronized block.